app/src/main/java/org/gnucash/android/db/BookDbHelper.java

```diff
@@ -65,64 +57,27 @@ public class BookDbHelper extends SQLiteOpenHelper {
 
     public BookDbHelper(Context context) {
         super(context, DatabaseSchema.BOOK_DATABASE_NAME, null, DatabaseSchema.BOOK_DATABASE_VERSION);
-        mContext = context;
     }
 
     @Override
     public void onCreate(SQLiteDatabase db) {
         db.execSQL(BOOKS_TABLE_CREATE);
 
-        if (mContext.getDatabasePath(DatabaseSchema.LEGACY_DATABASE_NAME).exists()) {
-            Timber.d("Legacy database found. Migrating to multibook format");
-            DatabaseHelper helper = new DatabaseHelper(GnuCashApplication.getAppContext(),
-                    DatabaseSchema.LEGACY_DATABASE_NAME);
-            SQLiteDatabase mainDb = helper.getWritableDatabase();
-            AccountsDbAdapter accountsDbAdapter = new AccountsDbAdapter(mainDb,
-                    new TransactionsDbAdapter(mainDb, new SplitsDbAdapter(mainDb)));
-
-            String rootAccountUID = accountsDbAdapter.getOrCreateGnuCashRootAccountUID();
+        Book book = new Book();
+        DatabaseHelper helper = new DatabaseHelper(GnuCashApplication.getAppContext(), book.getUID());
+        SQLiteDatabase mainDb = helper.getWritableDatabase(); //actually create the db
+        AccountsDbAdapter accountsDbAdapter = new AccountsDbAdapter(mainDb,
+                new TransactionsDbAdapter(mainDb, new SplitsDbAdapter(mainDb)));
 
-            Book book = new Book(rootAccountUID);
-            book.setActive(true);
-            insertBook(db, book);
-
-            String mainDbPath = mainDb.getPath();
+        String rootAccountUID = accountsDbAdapter.getOrCreateGnuCashRootAccountUID();
+        try {
+            accountsDbAdapter.close();
             helper.close();
-
-            File src = new File(mainDbPath);
-            File dst = new File(src.getParent(), book.getUID());
-            try {
-                MigrationHelper.moveFile(src, dst);
-            } catch (IOException e) {
-                String err_msg = "Error renaming database file";
-                Timber.e(e, err_msg);
-            }
-
-            migrateBackupFiles(book.getUID());
+        } catch (IOException ignore) {
         }
-
-        String sql = "SELECT COUNT(*) FROM " + BookEntry.TABLE_NAME;
-        SQLiteStatement statement = db.compileStatement(sql);
-        long count = statement.simpleQueryForLong();
-        if (count == 0) { //no book in the database, create a default one
-            Timber.i("No books found in database, creating default book");
```

Comment:
revert - don't create book each time use app

Reply:
onCreate() is called when the database is created, it's not called every time the app is opened.